gramineproject/gramine
# [LibOS,Pal/Linux-SGX] Don't use RBX as operand in inline asm

## Description of the changes 

It seems that in LLVM, the RBX register is reserved, and using it as inline assembly operand can produce unexpected results. In our case, Clang emitted incorrect code around the ENCLU instruction when compiling with ASAN and optimizations enabled.

To fix the problem, I'm moving the ENCLU instructions to separate functions defined in assembly files. I'm also updating one instance of RBX use in LibOS.

Fixes #391 (which describes one of the issues in #208).

## How to test this PR? 

Apart from Jenkins, you can compile Gramine with Clang with `--buildtype=debugoptimized` or `release` and ASAN enabled, and verify that the `fork_and_exec` test case passes. See also #391.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/398)
